src/tests/test_core_python.py

```diff
@@ -1,70 +1,99 @@
+from functools import cache
 from pathlib import Path
+from typing import Any
 
 import pytest
+import yaml
+from yaml import CLoader as Loader
 
 from pyodide_build.common import UNVENDORED_STDLIB_MODULES
 
 
-def test_cpython_core(python_test, selenium, request):
-
-    name, error_flags = python_test
-
+def filter_info(info: dict[str, Any], browser: str) -> dict[str, Any]:
+    info = dict(info)
     # keep only flags related to the current browser
     flags_to_remove = ["firefox", "chrome", "node"]
-    flags_to_remove.remove(selenium.browser)
-    for flag in flags_to_remove:
-        if "crash-" + flag in error_flags:
-            error_flags.remove("crash-" + flag)
+    flags_to_remove.remove(browser)
+    for browser in flags_to_remove:
+        for key in list(info.keys()):
```

Comment:
```suggestion
        for key in info:
```

Reply:
Don't you get in place modification errors from this?

Reply:
`RuntimeError: dictionary changed size during iteration`